plugins/sentinelone/komand_sentinelone/connection/connection.py

```diff
@@ -345,6 +346,66 @@ def get_events(self, params: dict, get_all_results: bool, event_type: str = None
 
         return insightconnect_plugin_runtime.helper.clean(self._call_api("GET", endpoint, params=params))
 
+    def update_analyst_verdict(self, incident_ids: list, analyst_verdict: str, _type: str) -> dict:
+        if _type == "threats":
+            endpoint = "threats/analyst-verdict"
+        else:
+            endpoint = "cloud-detection/alerts/analyst-verdict"
+
+        return self._call_api(
+            "POST",
+            endpoint,
+            {"filter": {"ids": incident_ids}, "data": {"analystVerdict": analyst_verdict}},
+        )
+
+    def update_incident_status(self, incident_ids: list, incident_status: str, _type: str) -> dict:
+        if _type == "threats":
+            endpoint = "threats/incident"
+        else:
+            endpoint = "cloud-detection/alerts/incident"
+
+        return self._call_api(
+            "POST",
+            endpoint,
+            {"filter": {"ids": incident_ids}, "data": {"incidentStatus": incident_status}},
+        )
+
+    def validate_incidents_exist(self, incident_ids: list, _type: str) -> list:
+        for incident_id in incident_ids:
+            response_data = self.get_incident(incident_id, _type).get("data")
+            if isinstance(response_data, list) and response_data.__len__() == 0:
+                self.logger.info(f"Incident {incident_id} was not found.")
+                incident_ids.remove(incident_id)
+
+        return incident_ids
+
```

Comment:
By function name - we are checking if it exists, so we would expect a boolean
But this function removes entries after checking existence.
A docstring comment explaining this would be helpful for code maintainability